pkg/tasks/tasks.go

```diff
@@ -253,6 +255,18 @@ func WithResources(t Tasks) Tasks {
 				Fn:     machinecontroller.WaitReady,
 				ErrMsg: "failed to wait for machine-controller",
 			},
+			{
+				Fn:          cloudinitsettings.Ensure,
+				ErrMsg:      "failed to create/configure cloud-init-settings namespace",
+				Description: "ensure cloud-init-settings namespace",
+				Predicate:   func(s *state.State) bool { return s.Cluster.AddonOperatingSystemManagerEnabled() },
+			},
+			{
+				Fn:          operatingsystemmanager.WaitReady,
```

Comment:
I have gut feeling that we can combine this with machine-controller's. Just pass them some parameters as target for waiting.

Reply:
We can but tbh I'd preferably keep this as a separate task. Eventually, OSM will be treated as a separate component/addon, we won't have to re-do stuff then.